x/session/keeper/session_hydrator.go

```diff
@@ -167,6 +168,14 @@ func (k Keeper) hydrateSessionSuppliers(ctx context.Context, sh *sessionHydrator
 
 	candidateSuppliers := make([]*sharedtypes.Supplier, 0)
 	for _, s := range suppliers {
+		// Exclude suppliers that are in an unbonding period.
+		// TODO_TECHDEBT: Suppliers that stake mid-session SHOULD NOT be included
+		// in the current session's suppliers list and must wait until the next one.
+		if s.UnstakeCommitSessionEndHeight != suppliertypes.SupplierNotUnstaking &&
```

Comment:
can we create a `func (s *Supplier) isUnbonding() bool` function as a helper?

Reply:
It would have to take a height argument.

```
func (s *Supplier) isUnbonding(queryHeight int64) bool
```

Reply:
@bryanchriswhite, I think there's a nuance here that needs to be considered given the value of `UnstakeSessionEndHeight`:
* `0`: never unstaked
* `!= 0 && < currentSessionEnd`: initiated an unstake but still active (serving relays)
* `> currentSessionEnd`: is unbonding and no longer active (serving relays)

I believe `IsUnbonding` should only indicate if the supplier initiated an unstake, and not whether it's sill active or not. This would maximize the reusablility of the method.

Given that, it's signature should be `IsUnbonding() bool` and should not take the height as argument.

We could still have a `IsActive(queryHeight) bool` that Indicates whether it should be included in sessions or not.